1.0/docs/devguide/templates.md

```diff
@@ -166,6 +168,79 @@ Changing a `manager.type` field should now cause the list to be re-sorted:
 
     this.set('employees.0.manager.type', 'engineer');
 
+#### Dynamic sort and filter changes
+
+The `observe` property lets you specify item sub-properties to 
+observe for filtering and sorting purposes. However, sometimes you want to
+dynamically change the sort or filter based on another unrelated value. In 
+this case, you can use a computed binding to _return_ a dynamic filter or 
+sort function when one or more dependent properties changes.
+
+    <dom-module id="employee-search">
+
+      <template>{% raw %}
+        <input value="{{searchString::change}}">
+        <template is="dom-repeat" items="{{employees}}" as="employee"
+            filter="{{matches(searchString)}}">
+            <div>{{employee.lastname}}, {{employee.firstname}}</div>
+        </template>{% endraw %}
+      </template>
+
+      <script>
+        Polymer({
+          is: "employee-search",
+          matches: function(string) {
+            if (!string) {
+              // if no search string, show all records
+              return function() { return true; }
+            } else {
+              // return a filter function for the current search string
+              string = string.toLowerCase();
```

Comment:
Not exactly. But your comment points out to me that `matches` is a terrible function name here, because the function doesn't _match_, anything, it _returns_ a filter function. So with a slight rephrasing: 

          filterFactory: function(searchString) {
            if (!searchString) {
              // if no search string, return a function that matches all records
              return function() { return true; }
            } else {
              // return a filter function for the current search string
              searchString = searchString.toLowerCase();
              return function(employee) {
                var first = employee.firstname.toLowerCase();
                var last = employee.lastname.toLowerCase();
                return first.includes(searchString) ||
                    last.includes(searchString);
              }
            }
          },

So what's happening here? Whenever `searchString` changes, the computed binding `{{filterFactory(searchString}}` is re-evaluated. It returns a Function object, which is assigned as the value of the `dom-repeat`'s `filter` property.

Changing the `filter` property causes the `dom-repeat` to re-filter its items (by testing each one with the new filter function).